File: src/mobile/lib/app/utils/models/notificacoes/notificacao_model.dart
class NotificacaoModel {
  NotificacaoModel({
    required this.titulo,
    required this.mensagem,
    this.imageUrl,
  });
  late final String titulo;
  late final String mensagem;
  late final String? imageUrl;

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['titulo'] = titulo;
    data['mensagem'] = mensagem;
    data['imageUrl'] = imageUrl;
    return data;
  }

  NotificacaoModel.fromJson(Map<String, dynamic> json) {
    titulo = json['titulo'];
    mensagem = json['mensagem'];
    imageUrl = json['imageUrl'];
  }
}

class NotificacaoResponseModel {
  NotificacaoResponseModel({required this.id});
  late final String id;

  NotificacaoResponseModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
  }
}

File: src/services/Catalogo/Catalogo.API/Models/NotificacaoModel.cs
using System.ComponentModel.DataAnnotations;

namespace Catalogo.API.Models
{
  public class NotificacaoModel
  {
    [Required(ErrorMessage = "Título da notificação está vazio.")]
    public string Titulo { get; set; } = null!;

    [Required(ErrorMessage = "Mensagem da notificação está vazio.")]
    public string Mensagem { get; set; } = null!;

    public string? ImageUrl { get; set; }    
  }

  public record NotificacaoResponseModel
  {
    public string Id { get; set; }

    public NotificacaoResponseModel(string id)
    {
      Id = id;
    }
  }
}